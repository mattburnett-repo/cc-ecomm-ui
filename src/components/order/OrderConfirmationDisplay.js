export default function OrderConfirmationDisplay ( props ) {
    const { data } = props;
    
    return (
        <div>
            <div role="presentation" aria-label="confirmation-info">
                <div role="presentation" aria-label="order-amount-date">
                    Your order for {data.total_amount} has been placed on {data.order_date}
                </div>
                <div role="presentation" aria-label="order-payment">
                    Payment in the amount of {data.total_amount} was made using {data.payment_type}
                </div>
                <div role="presentation" aria-label="order-number">
                    The order number is {data.order_id}
                </div>
                <div role="presentation" aria-label="order-delivery">
                    Order will be delivered to {data.delivery_address}
                </div>
            </div>
            <button aria-label="home">Home</button>
            <button aria-label="logout">Log Out</button>
        </div>
    )
}