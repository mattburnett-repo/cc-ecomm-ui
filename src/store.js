import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;