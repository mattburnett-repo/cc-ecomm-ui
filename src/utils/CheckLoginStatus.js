
import { useSelector } from 'react-redux'
import { useHistory } from 'react-router-dom'

import { selectIsAuthenticated } from '../features/auth/authSlice'

export default function CheckLoginStatus () {
    // eslint-disable-next-line
    const isLoggedIn = useSelector(selectIsAuthenticated);

    const history = useHistory()

    if(!isLoggedIn) {
        history.push('/login')

        return false
    } else {
        return true;
    }
}        
