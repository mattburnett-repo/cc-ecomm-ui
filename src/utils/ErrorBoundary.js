import React from 'react'

export default class ErrorBoundary extends React.Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
  
    static getDerivedStateFromError(error) {
      // Update state to trigger fallback UI
      return { hasError: true };
    }
  
    componentDidCatch(error, errorInfo) {
        // Log error or send logging data to log management tool
        //   logErrorToMyService(error, errorInfo);
        console.log('ErrorBoundary ', errorInfo)
    }
  
    render() {
      if (this.state.hasError) {
        // Render fallback UI
        return <h1>There was an error. Sorry about that; we're working on it.</h1>;
      }
  
      return this.props.children; 
    }
  }

