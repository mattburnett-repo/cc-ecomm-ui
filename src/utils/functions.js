
// USAGE: formatAsCurrency.format(3.005); // "3.01"
const formatAsCurrency = new Intl.NumberFormat('en-US', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
});

const formatDate =   new Intl.DateTimeFormat("en-US", {
    year: "numeric",
    month: "long",
    day: "2-digit"
});

module.exports = {
    formatAsCurrency: formatAsCurrency,
    formatDate: formatDate
}