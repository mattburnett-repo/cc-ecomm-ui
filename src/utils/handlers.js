

// handler function declarations

// function handleAddToCart(e){ 
//     alert('handleAddToCart inside utils product_id: ' + e.target.id + ' quantity: ' + e.quantity)
//     // {() => dispatch(addItemToCurrentCart(id))}
//     // can't useDispatch here.
// } 

function handleCheckout(e){ 
    alert('handleCheckout inside utils ')
} 

function handleSearch(e){
    e.preventDefault()
    // let searchTerms = e.target.value
    alert('handleSearch for search term/s ' + e.target.searchTerms.value + '. \nneed to figure out how to get dispatch() to work from here')
}

function handleOrderClick(e) {
    e.preventDefault()

    let order_id = e.target.order_id.value
    alert('go to order detail component ' + order_id)
}

function handleHomeClick() {
    console.log('handleHomeClick')
}
function handleShowCartClick() {
    console.log('handleShowCartClick')
}
function handleCheckoutClick() {
    console.log('handleCheckoutClick')
}
function handleCancelClick() {
    console.log('handleCanceClick')
}


module.exports = {
    handleHomeClick: handleHomeClick,
    handleShowCartClick: handleShowCartClick,
    handleCheckoutClick: handleCheckoutClick,
    handleCancelClick: handleCancelClick,
    // handleAddToCart: handleAddToCart,
    handleCheckout: handleCheckout,
    handleSearch: handleSearch,
    handleOrderClick: handleOrderClick
} // end module.exports