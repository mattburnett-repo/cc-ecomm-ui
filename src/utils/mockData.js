// https://css-tricks.com/writing-tests-for-react-applications-using-jest-and-enzyme/

module.exports = {
    // someday I will figure out how to do this correctly...
    // getUserData: jest.fn(() => {
    //   return Promise.resolve({
    //   data: [
    //     {
    //       id: 1,
    //       username: 'asdf',
    //       email: 'asdf@asdf.com',
    //       password: 'asdf'
    //     }
    //   ]
    //   })
    // }),
    mockCartsData:[
        {
          "cart": {
              "cart_id": 1,
              "user_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "hedgehog",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "maine coon cat",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "bulldog",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "bunny",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ],
              "cart_total_price": {
                  "sum": 30,
                  "cart_id": 1
              }
          }
      }
    ],
    mockCartData: {
      cart: {
        cart_id: 1,
        user_id: 1,
        cart_items: [
          {
            product_id: 1,
            product_name: 'hedgehog',
            product_price: 1,
            product_quantity: 1,
            product_total_price: 1
          },
          {
            product_id: 2,
            product_name: 'maine coon cat',
            product_price: 2,
            product_quantity: 2,
            product_total_price: 4
          },
          {
            product_id: 3,
            product_name: 'bulldog',
            product_price: 3,
            product_quantity: 3,
            product_total_price: 9
          },
          {
            product_id: 4,
            product_name: 'bunny',
            product_price: 4,
            product_quantity: 4,
            product_total_price: 16
          }
      ],
      cart_total_price: { sum: 30, cart_id: 1 }
    }},
    mockOrderData: [{
      "order": {
          "cart": {
              "cart_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "product_name_01",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "product_name_02",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "product_name_03",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "product_name_04",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ]
          },
          "user_id": 1,
          "order_id": 1,
          "order_date": null,
          "total_price": {
              "sum": 30,
              "cart_id": 1
          }
        }
      }],
    mockOrderConfirmationData:
      {
        "order_id": 1,
        "total_amount": 1234,
        "order_date": '12-Dez-2021',
        "payment_type": 'Deutschmarks',
        "delivery_address": '123 Katzendamm, 90210 Berlin DE'
      },
    mockProductCategories: [
        // {
        //   "category_id": 0,
        //   "description": "Please select a product category"
        // },
        {
          "category_id": 4,
          "description": "Auto"
        },
        {
            "category_id": 3,
            "description": "Health"
        },
        {
            "category_id": 1,
            "description": "Pets"
        },
        {
            "category_id": 2,
            "description": "Tech"
        }
      ],
    mockProduct:[
        {
          "id": 2,
          "category_id": 2,
          "name": "product_name_02",
          "description": "product_desc_02",
          "image_url": "https://cdn.pixabay.com/photo/2015/03/27/13/16/maine-coon-694730_1280.jpg",
          "price": "2.00"
        },      
      ],
    mockProducts: [
        {
            "id": 1,
            "category_id": 1,
            "name": "product_name_01",
            "description": "product_desc_01",
            "image_url": "https://pixabay.com/images/id-71282/",
            "price": "1.00"
        },
        {
            "id": 2,
            "category_id": 2,
            "name": "product_name_02",
            "description": "product_desc_02",
            "image_url": "http://www.image.com/item/2",
            "price": "2.00"
        },
        {
            "id": 3,
            "category_id": 3,
            "name": "product_name_03",
            "description": "product_desc_03",
            "image_url": "http://www.image.com/item/3",
            "price": "3.00"
        },
        {
            "id": 4,
            "category_id": 4,
            "name": "product_name_04",
            "description": "product_desc_04",
            "image_url": "http://www.image.com/item/4",
            "price": "4.00"
        }
    ],
    mockUserData:[
      {
        "id": 4,
        "user_name": "asdf",
        "password": "$2b$10$OsTX1L59rYgsmAnsrJsXoO3RH3RSfLgEF53YBB3723B9eoELtUa.a",
        "email": "asdf@asdf.com",
        "google_id": null,
        "google_display_name": null,
        "google_first_name": null,
        "google_last_name": null,
        "google_image": null
      }
    ]
  } // end module.exports