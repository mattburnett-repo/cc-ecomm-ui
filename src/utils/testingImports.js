// export default function TestingImports() {
//     return (
//         import { render, screen, fireEvent } from '@testing-library/react';
//         import renderer from 'react-test-renderer';
//         import userEvent from '@testing-library/user-event';

//         import { Provider } from 'react-redux';
//         import store from '../../store'
//     )
// }