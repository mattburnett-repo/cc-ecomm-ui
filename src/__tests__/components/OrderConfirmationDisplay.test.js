// 20211217: Wireframe https://wireframe.cc/KF5blD

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux'
import store from '../../store'

import OrderConfirmationDisplay from '../../components/order/OrderConfirmationDisplay';

import { mockCurrentOrderData, mockCurrentPaymentData, mockCurrentAddressData } from '../../utils/mockData'

describe('<OrderConfirmationDisplay data={mockData} /> component tests', () => {

    test.todo('update OrderConfirmation tests to reflect updated design / data')
    
    // beforeEach(() => render(<Provider store={store}>
    //                             <OrderConfirmationDisplay 
    //                                 currentOrder={mockCurrentOrderData} 
    //                                 currentPayment={mockCurrentPaymentData}
    //                                 currentAddress={mockCurrentAddressData} />
    //                         </Provider>));

    // it('should render confirmation info', () => {
    //     screen.getByRole('presentation', {name: /confirmation-info/i});
    // });
    // it('should render a home button', () => {
    //     screen.getByRole('button', {name: /home/i});
    // });
    // it('should render a logout button', () => {
    //     screen.getByRole('button', {name: /logout/i});
    // });

    // it('creates a snapshot', () => {
    //     const tree = renderer.create(<Provider store={store}>
    //                                     <OrderConfirmationDisplay 
    //                                         currentOrder={mockCurrentOrderData} 
    //                                         currentPayment={mockCurrentPaymentData}
    //                                         currentAddress={mockCurrentAddressData} />
    //                                 </Provider>).toJSON();
    //     expect(tree).toMatchSnapshot();  
    // });
}); // end component
