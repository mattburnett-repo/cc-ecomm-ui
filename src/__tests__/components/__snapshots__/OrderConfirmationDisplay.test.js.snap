// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderConfirmationDisplay data={mockData} /> component tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="confirmation-info"
    role="presentation"
  >
    <div
      aria-label="order-amount-date"
      role="presentation"
    >
      Your order for 
      1234
       has been placed on 
      12-Dez-2021
    </div>
    <div
      aria-label="order-payment"
      role="presentation"
    >
      Payment in the amount of 
      1234
       was made using 
      Deutschmarks
    </div>
    <div
      aria-label="order-number"
      role="presentation"
    >
      The order number is 
      1
    </div>
    <div
      aria-label="order-delivery"
      role="presentation"
    >
      Order will be delivered to 
      123 Katzendamm, 90210 Berlin DE
    </div>
  </div>
  <button
    aria-label="home"
  >
    Home
  </button>
  <button
    aria-label="logout"
  >
    Log Out
  </button>
</div>
`;
